dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1836,27 +1849,6 @@ Spdp::match_authenticated(const DCPS::RepoId& guid, DiscoveredParticipantIter& i
     }
   }
 
-  // Must unlock when calling into part_bit() as it may call back into us
-  ACE_Reverse_Lock<ACE_Thread_Mutex> rev_lock(lock_);
-
-  DDS::InstanceHandle_t bit_instance_handle = DDS::HANDLE_NIL;
-#ifndef DDS_HAS_MINIMUM_BIT
-  if (!secure_part_user_data()) { // else the user data is assumed to be blank
-    DCPS::ParticipantBuiltinTopicDataDataReaderImpl* bit = part_bit();
-    if (bit) {
-      DDS::ParticipantBuiltinTopicData pbtd = partBitData(iter->second.pdata_);
-      ACE_GUARD_RETURN(ACE_Reverse_Lock<ACE_Thread_Mutex>, rg, rev_lock, false);
-      bit_instance_handle =
-        bit->store_synthetic_data(pbtd, DDS::NEW_VIEW_STATE);
-      rg.release();
-      iter = participants_.find(guid);
-      if (iter == participants_.end()) {
-        return false;
-      }
-    }
-  }
-#endif /* DDS_HAS_MINIMUM_BIT */
-
```

Comment:
Why remove this part? Isn't this when a secure participant without secure user data would get the participant bit data or is this happening somewhere else?

Reply:
Secure user data dictates what you send, not what you receive.  Reception-wise, a secure participant should only trust the data received from the secure/reliable SPDP writer.  The code in `handle_participant_data` has now been expanded to write to the BIT when this data arrives the first time.